### ISSUE TITLE
implement "implements"

### ISSUE THREAD

**Issue by aszs** [codemix/flow-runtime#127]
This is a:
 - [ x ] Feature Request

Which concerns:

 - [ x ] flow-runtime
 - [ x ] babel-plugin-flow-runtime

### What is the current behaviour?

The interfaces a class implements are missing from its type annotation.E.g.:

```
interface Serializable {
  serialize(): string;
}

class Foo implements Serializable {
  serialize() { return '[Foo]'; } // Works!
}
```
transforms to
```
@t.annotate(t.class('Foo', t.method('serialize')))
class Foo {
  constructor() {
    Serializable.assert(this);
  }

  serialize() {
    return '[Foo]';
  } // Works!
}
```

### What is the expected behaviour?

The interfaces declared by "implements" would be included in the type annotation.

### Which package versions are you using?

Example is from whatever is running here: https://codemix.github.io/flow-runtime/#/try

**Comment by fabiandev** [comment]
I was also experiencing this behavior and was working around this issue, by defining an `IntersectionType`, containing class members as `ObjectType` and all interface references, and calling `unwrap()` on it in place. This is definitely not ideal and it may be of advantage to first investigate #123 and then allowing to pass a merged `ObjectType`, containing members of the class and its implemented interface.

**Comment by aszs** [comment]
For my needs passing in the referenced interface types is enough, for my use case I just need to know which interfaces the class explicitly declares that it implements. Sounds like your use case is more complicated.

**Comment by fabiandev** [comment]
You're totally right, it just needs to verify the signatures of the types, which it obviously doesn't.
In any case, I think you would need to annotate the class method as well.
